ddl/reorg.go

```diff
@@ -479,15 +489,16 @@ func getReorgInfo(d *ddlCtx, t *meta.Meta, job *model.Job, tbl table.Table) (*re
 		failpoint.Inject("errorUpdateReorgHandle", func() (*reorgInfo, error) {
 			return &info, errors.New("occur an error when update reorg handle")
 		})
-		err = t.UpdateDDLReorgHandle(job, start, end, pid)
+		err = t.UpdateDDLReorgHandle(job, start, end, pid, elements[0])
 		if err != nil {
 			return &info, errors.Trace(err)
 		}
 		// Update info should after data persistent.
 		job.SnapshotVer = ver.Ver
+		element = elements[0]
 	} else {
 		var err error
-		start, end, pid, err = t.GetDDLReorgHandle(job, tbl.Meta().IsCommonHandle)
+		element, start, end, pid, err = t.GetDDLReorgHandle(job, tbl.Meta().IsCommonHandle)
```

Comment:
When a DDL job is running then job.SnapshotVer is not 0. So when TiDB upgrade and the owner changed this will cause `t.GetDDLReorgHandle` return an error, and the DDL job will fail.
So I think we should inform end user do not upgrade TiDB when running DDL SQL.

Reply:
Sorry, metadata has been adjusted before, so there is no need to do special treatment here. Now, special treatment of metadata has been removed, so special treatment is needed here. I did it and add tests for it.